contribs/gmf/src/directives/layertree.js

```diff
@@ -975,16 +977,19 @@ gmf.LayertreeController.prototype.zoomToResolution = function(treeCtrl) {
 /**
  * Collect and return all ids of this layer node and all child nodes as well.
  * @param {GmfThemesNode} node Layer tree node.
+ * @param {boolean=} opt_editable Whether the node needs to be editable to
+ *     have its id returned.
  * @return {Array.<number|string>} Layer names.
  */
-gmf.LayertreeController.getLayerNodeIds = function(node) {
+gmf.LayertreeController.getLayerNodeIds = function(node, opt_editable) {
+  var editable = opt_editable === true;
   var ids = [];
   var children = node.children || node;
   if (children && children.length) {
     children.forEach(function(childNode) {
       ids = ids.concat(gmf.LayertreeController.getLayerNodeIds(childNode));
```

Comment:
When `opt_editable` is true, you want to get the ids of the node which are editable and only them. Calling `getLayerNodeIds` for each child of the current node without the "editable" argument will retrieve the ids of the children even if they're not editable.
I may be wrong but you'll get the editable nodes for the first level but all the nodes for level greater than 2.